spec/enumerables_spec.rb

```diff
@@ -0,0 +1,211 @@
+require './enumerable'
+
+RSpec.describe Enumerable do
+  let(:testArray) { [1, 2, 3, 4, 5] }
+  describe '#my_each' do
+    it 'Return a valid enumerable' do
+      expect(testArray.my_each).not_to be_a(String)
+    end
+
+    it 'Return a valid enumerable' do
+      expect(testArray.my_each { |element| element }).not_to eql(nil)
+    end
+
+    it 'Return an array equalt to the original [1,2,3,4,5]' do
+      expect(testArray.my_each { |element| element }).to eql(testArray)
+    end
+
+    it 'Returns a valid Enumerable when no vlock is given' do
+      expect(testArray.my_each).to be_a(Enumerable)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Return a  valid enumerable if no block given' do
+      expect(testArray.my_each_with_index).to be_a(Enumerable)
+    end
+
+    it 'Returns an Valid Enumerable' do
+      expect(testArray.my_each_with_index).not_to be_a(String)
+    end
+    let(:expected_testArray) { [] }
+
+    it 'Returns a new array substracting 1 from each array' do
+      testArray.my_each_with_index { |_x, i| expected_testArray[i] = testArray[i] - 1 }
+      expect(expected_testArray).to eql([0, 1, 2, 3, 4])
+    end
+
+    it 'Does block instructions on each element (negative)' do
+      testArray.my_each_with_index { |_x, i| expected_testArray[i] = testArray[i] - 1 }
+      expect(expected_testArray).not_to eql(testArray)
+    end
+  end
+
+  describe '#my_select' do
+    let(:expected_testArray) { testArray.my_select(&:even?) }
+
+    it 'Expected all even numbers from [1,2,3,4,5]' do
+      expect(expected_testArray).to eql([2, 4])
+    end
+
+    it 'Expected all even numbers from [1,2,3,4,5' do
+      expect(expected_testArray).not_to eql([])
+    end
+
+    let(:tempArr) { testArray.my_select }
+
+    it 'Return valid enumerable when no block is passed' do
+      expect(tempArr).not_to be_a(Integer)
+    end
+  end
+
+  describe '#my_all?' do
+    let(:tempArr) { testArray.my_all?(&:even?) }
+
+    it 'Return false if not all elements fill criteria' do
+      expect(tempArr).to eql(false)
+    end
+
+    it 'Returns false if not all elements fill criteria (negative)' do
+      expect(tempArr).not_to eql(true)
+    end
+
+    let(:expected_testArray) { testArray.my_all? { |x| x > 0 } }
+
+    it 'Returns true if all elements fill criteria' do
+      expect(expected_testArray).to eql(true)
+    end
+
+    it 'Returns true if all elements fill criteria (negative)' do
+      expect(expected_testArray).not_to eql(false)
+    end
+
```

Comment:
This test is actually the same(both positive) test, but you can do this way
#### Positive
    let(:expected_testArray) { testArray.my_all? { |x| x > 0 } }
    expect(expected_testArray).to eql(true)
#### Negative
`let(:expected_Array) { testArray.my_all? { |x| x < 0 } }`
`expect(expected_Array).to eql(false)`